File: backend/mscliente/src/main/java/mscliente/mscliente/controllers/ClienteController.java
package mscliente.mscliente.controllers;

import java.util.List;

import org.modelmapper.ModelMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import mscliente.mscliente.DTO.ClienteDTO;
import mscliente.mscliente.model.Cliente;
import mscliente.mscliente.services.ClienteService;

@RestController
@RequestMapping("/clientes")
public class ClienteController {

    @Autowired
    private ClienteService service;


    @PostMapping
    public ResponseEntity <ClienteDTO> createCliente(@RequestBody ClienteDTO cliente) {
        ClienteDTO saved = service.saveCliente(cliente);
        return ResponseEntity.status(HttpStatus.CREATED).body(saved);
    }

    @GetMapping
    public List<Cliente> listar() {
        return service.listarClientes();
    }
}

File: backend/cliente-service/src/main/kotlin/br/ufpr/dac/cliente_service/resource/ClienteController.kt
package br.ufpr.dac.cliente_service.resource

import org.springframework.http.ResponseEntity
import utils.dto.ClienteInputDTO
import utils.dto.ClienteOutputDTO
import org.springframework.web.bind.annotation.*
import org.springframework.http.HttpStatus
import org.springframework.beans.factory.annotation.Value

@RestController
@RequestMapping("/clientes")
@CrossOrigin(origins = ["\${gateway.url}"])
class ClienteController(private val service: ClienteService) {

    @GetMapping
    fun getClientes(): ResponseEntity<List<ClienteOutputDTO>>{
        val clientes = service.getAllClientes()
        return ResponseEntity.ok().body(clientes)
    }
    
    @DeleteMapping("/{id}")
    fun deleteCliente(@PathVariable id: Long): ResponseEntity<ClienteOutputDTO>{
      val deactivatedCliente = service.deactivateCliente(id)
      return ResponseEntity.ok(deactivatedCliente)
    }

    @GetMapping("/{id}")
    fun getClienteById(@PathVariable id: Long) : ResponseEntity<ClienteOutputDTO>{
      val cliente = service.getClienteByID(id)
      return ResponseEntity.ok(cliente)
    }

    @PostMapping
    fun createCliente(@RequestBody cliente: ClienteInputDTO): ResponseEntity<ClienteOutputDTO>{
      val savedCliente =  service.saveCliente(cliente)
      return ResponseEntity.status(HttpStatus.CREATED).body(savedCliente)
    }
    
    @PutMapping("/{id}")
    fun updateCliente(@PathVariable id: Long, @RequestBody clienteDTO: ClienteInputDTO): ResponseEntity<ClienteOutputDTO>{
      val updatedCliente = service.updateCliente(id, clienteDTO)
      return ResponseEntity.ok().body(updatedCliente)
    }
  }
